core/src/main/java/nl/nn/adapterframework/pipes/CsvParserPipe.java

```diff
@@ -84,34 +87,36 @@ public void configure() throws ConfigurationException {
 
 	@Override
 	public PipeRunResult doPipe(Message message, PipeLineSession session) throws PipeRunException {
-		try (MessageOutputStream target=getTargetStream(session)) {
+		try {
+			File tempFile = FileUtils.createTempFile();
 			try (Reader reader = message.asReader()) {
-				try (SaxDocumentBuilder document = new SaxDocumentBuilder("csv", target.asContentHandler(), isPrettyPrint())) {
+				try (SaxDocumentBuilder document = new SaxDocumentBuilder("csv", Files.newBufferedWriter(tempFile.toPath()), isPrettyPrint())) {
 					CSVParser csvParser = format.parse(reader);
-					for (CSVRecord record : csvParser) {
-						try (SaxElementBuilder element = document.startElement("record")) {
-							for(Entry<String,String> entry:record.toMap().entrySet()) {
-								String key = entry.getKey();
-								if(getHeaderCase() != null) {
-									key = getHeaderCase()==HeaderCase.LOWERCASE ? key.toLowerCase() : key.toUpperCase();
-								}
-								element.addElement(key, entry.getValue());
-							}
-						} catch (SAXException e) {
-							throw new PipeRunException(this, "Exception caught at line ["+record.getRecordNumber()+"] pos ["+record.getCharacterPosition()+"]", e);
-						}
+					for (CSVRecord csvRecord : csvParser) {
+						processCsvRecord(csvRecord, document);
 					}
 				}
 			}
-			return target.getPipeRunResult();
-		} catch (Exception e) {
-			if (e instanceof PipeRunException) {
-				throw (PipeRunException)e;
-			}
+			return new PipeRunResult(getSuccessForward(), PathMessage.asTemporaryMessage(tempFile.toPath(), message.getCharset()));
```

Comment:
Het is niet handig hier de charset over te nemen. Die gebruikt je bij het lezen. Hier geef je (altijd) utf-8 data terug (want XML).

Reply:
Oke, ik haal hem weg. XML is idd de output, zie ik.